### PR TITLE
Add CSPOrgID and EntitlementAccountNumber to telemetry options in the config 

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR adds `cspOrgID` and `entitlementAccountNumber` to the telemetry options in the config
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Unit tests and CI checks passed

### Release note

```release-note
Add CSPOrgID and EntitlementAccountNumber to telemetry options in the config 
```

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -83,6 +83,13 @@ func GetEnvConfigurations() map[string]string
 func GetEdition() (string, error)
 func SetEdition(val string) (err error)
 
+// Telemetry APIs
```
Comment (prkalle) [inline docs/config.md:86]: I hope adding the telemetry APIs in the docs is fine, though these current APIs are used by CLI and telemetry plugin only.

**Review by anujc25 — APPROVED** [review]
LGTM

File: config/cli_options_telemetry_test.go [hunk 2]
```diff
@@ -20,13 +20,21 @@ func TestSetGetTelemetryOptions(t *testing.T) {
...
+		EntitlementAccountNumber: "fake-entitlement-account-number",
 	}
 
 	telOptionsUpdate := &configtypes.TelemetryOptions{
 		Source: "/fake/path/updated",
 	}
 
+	telOptionsUpdate2 := &configtypes.TelemetryOptions{
+		Source:                   "/fake/path/updated-2",
+		CSPOrgID:                 "fake-csp-org-id-2",
```
Comment (mpanchajanya) [inline config/cli_options_telemetry_test.go:34]: fyi. if empty "" fields are passed to update. APIs will not update the config if value is already present.  We need to use replace patch strategy on those specific fields

**Review by mpanchajanya — APPROVED** [review]
lgtm! fyi. if empty "" fields are passed to update. APIs will not update the config if value is already present. We need to use replace patch strategy on those specific fields

**Comment by prkalle** [comment]
> lgtm! fyi. if empty "" fields are passed to update. APIs will not update the config if value is already present. We need to use replace patch strategy on those specific fields

Thanks! I think users should do a GET, followed by DELETE and then SET( the specific field to empty and rest unchanged) inorder to set the filed to empty(delete).

**Review by marckhouzam — APPROVED** [review]
LGTM

**Pull request merged**